app/src/main/kotlin/com/adesso/movee/scene/tvshowdetail/TvShowDetailViewModel.kt

```diff
@@ -29,13 +29,8 @@ class TvShowDetailViewModel @Inject constructor(
 
     fun fetchTvShowDetail(id: Long) {
         if (_tvShowDetails.value == null) {
-            bgScope.launch {
-                val tvShowDetailResult =
-                    fetchTvShowDetailUseCase.run(FetchTvShowDetailUseCase.Params(id))
-
-                onUIThread {
-                    tvShowDetailResult.either(::handleFailure, ::postTvShowDetail)
-                }
+            fetchTvShowDetailUseCase.invoke(viewModelScope, FetchTvShowDetailUseCase.Params(id)) {
+                it.either(::handleFailure, ::postTvShowDetail)
```

Comment:
postTvShowDetail() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postTvShowDetail()